packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -160,6 +162,24 @@ export class DocumentDeltaConnection
 					this.emit(event, ...args);
 				});
 			}
+
+			// log latency every minute
+			this.trackLatencyTimer = setInterval(() => {
```

Comment:
Should we be setting this multiple times?
Also, should we only care about this latency tracking when someone listens to the delta connection's "pong" event?

Reply:
hmmm what do you mean multiple times? my understanding is whatever new listener is triggered, we will run this loop.
for the second question, do you mean we should track latency for other events as well?

Reply:
From my understanding, a `setInterval` will continue to run until you call `clearInterval` on it. Thus, you only need to set the `this.trackLatencyTimer` once (whereas it's currently being set every time someone listens to an event on the delta connection object).
Not 100% sure, but this may cause an issue of older interval IDs not being cleared since the `this.trackLatencyTimer` value is being overwritten.

My second question was moreso towards when do we want to do this latency analysis? Currently, it's being done whenever someone listens to any new event on the document connection object. We could add some handling to only do this latency tracking when someone explicitly listens to the `"pong"` event. Not 100% necessary, but worth asking.
```
if (!this.trackedListeners.has(event)) {
	if (event === "pong") {
		this.trackLatencyTimer = setInterval(() => {
			...
		}, 1000 * 60);
	}
	this.addTrackedListener(event, (...args: any[]) => {
		this.emit(event, ...args);
	});
}
```

Reply:
Kian is correct - we should start a timer only if registration is for "pong" event.